### PR TITLE
fix softmax shape

### PR DESCRIPTION
**Pull request by roywei**
https://github.com/apache/incubator-mxnet/pull/14869 fix nightly test failure after unknown dim changed from 0 to -1 in MXNet.
This fix softmax shape check in keras after the change.
requires https://github.com/apache/incubator-mxnet/pull/14869 to go into nightly pip packages to pass CI.

### CODE REVIEW

**Comment by roywei** [comment]
@sandeep-krishnamurthy could you help take a look? would like this to go in before MXNet 1.5.0 release.
Now MXNet backend has more strick partial shape inference. So at this time, softmax shape is not inferred yet and should not go into the logic of ndim==1

**Comment by sandeep-krishnamurthy** [comment]
I am concerned with too many specialization for MXNet backend.

**Comment by roywei** [comment]
Thanks for the review, agree and created issue https://github.com/awslabs/keras-apache-mxnet/issues/233
